File: New folder/datadict_mapping.py
import json
from pathlib import Path

DATA_PATH = Path(__file__).resolve().parent / "dictionary.json"
with DATA_PATH.open("r", encoding="utf-8") as f:
    datadict = json.load(f)


def map_statN(code):
    return datadict["StatN"].get(str(code), "Unknown")


def map_statC(code):
    return datadict["StatC"].get(str(code), "Unknown")


def map_slot(slot):
    return datadict["slotMeaning"].get(slot, "Unknown")


def map_ID1(code):
    return datadict["ID1Type"].get(code, "Unknown")


def map_ID2(code):
    return datadict["ID2Type"].get(code, "Unknown")

File: datadict_mapping.js
import datadict from "./backend/utils/dictionary.json" assert { type: "json" };

export function mapStatN(code) {
  return datadict.StatN[String(code)] || "Unknown";
}

export function mapStatC(code) {
  return datadict.StatC[String(code)] || "Unknown";
}

export function mapSlot(slot) {
  return datadict.slotMeaning[slot] || "Unknown";
}

export function mapID1(code) {
  return datadict.ID1Type[code] || "Unknown";
}

export function mapID2(code) {
  return datadict.ID2Type[code] || "Unknown";
}
